packages/overture-schema-system/src/overture/schema/system/model_constraint/radio_group.py

```diff
@@ -0,0 +1,155 @@
+from collections.abc import Callable
+from types import NoneType, UnionType
+from typing import Annotated, Any, Union, get_args, get_origin
+
+from pydantic import BaseModel, ConfigDict
+from pydantic.json_schema import JsonDict
+from typing_extensions import override
+
+from .._json_schema import get_static_json_schema, put_one_of
+from .model_constraint import FieldGroupConstraint, apply_alias
+
+
+def radio_group(*field_names: str) -> Callable[[type[BaseModel]], type[BaseModel]]:
+    """
+    Decorates a Pydantic model class with a constraint requiring that exactly one field in a group
+    of `bool` fields has the value `True`.
+
+    This function is the decorator version of the `RadioGroupConstraint` class.
+
+    Historical node for nerds: the term radio group, meaning a group of radio buttons, was first
+    used in software user interfaces design as an analogy to the mechanical push buttons on the
+    physical home and automobile radio sets of the 1950s-1970s. These radio sets had preset station
+    buttons where pushing one button would physically pop out any other buttons, ensuring only one
+    station could be selected at a time.
```

Comment:
My first reaction when I saw the decorator in action was (yes, knowing that it's an analogy) "huh, using UI to describe this is weird," immediately followed by "but that's so much clearer than trying to communicate the effect" (which is what I was struggling with when I was first modeling this). Nice use of cross-domain terms.